pkg/tagger/tagstore.go

```diff
@@ -74,11 +80,24 @@ func (s *tagStore) processTagInfo(info *collectors.TagInfo) error {
 		}
 		s.store[info.Entity] = storedTags
 
+		eventType = EventTypeAdded
+
 		storedEntities.Inc()
 	}
 
 	updatedEntities.Inc()
 
+	err := updateStoredTags(storedTags, info)
+	if err != nil {
+		return err
+	}
+
+	s.notifySubscribers(info.Entity, storedTags, eventType)
```

Comment:
It seems to me that without the lock on the store, events can be sent out of order, and order is important. For example, an update shouldn't be sent after a delete, and without the lock there's no way to guarantee that order. It's also important when a new subscriber is being added, since it needs to read the store, register the channel, and only then it can start receiving events. Without the lock, I don't think we can preserve that order.

On channels being full: yes, it's quite possible, but I don't whether that will become a problem. I just made a change to alleviate that a bit (instead of sending single events, we group them in a list and send that), but even then it's not guaranteed. I'm not sure what the behavior should be in that case, but I'm thinking that whenever the channel is full it should be closed, which would end the stream and signal to the client to start a fresh new one. I'm happy to hear thoughts on that.